web/packages/shared/components/AccessRequests/NewRequest/RequestCheckout/RequestCheckout.tsx

```diff
@@ -206,18 +218,59 @@ export function RequestCheckout<T extends PendingListItem>({
     isResourceRequest &&
     selectedResourceRequestRoles.length < 1;
 
-  const submitBtnDisabled =
-    pendingAccessRequests.length === 0 ||
-    createAttempt.status === 'processing' ||
-    isInvalidRoleSelection ||
-    (fetchResourceRequestRolesAttempt.status === 'failed' &&
-      hasUnsupporteKubeResourceKinds) ||
-    fetchResourceRequestRolesAttempt.status === 'processing';
+  const submitBtnDisabled = useMemo(() => {
+    if (
+      pendingAccessRequests.length === 0 ||
+      createAttempt.status === 'processing' ||
+      isInvalidRoleSelection
+    )
+      return true;
+    if (
+      fetchResourceRequestRolesAttempt.status === 'failed' &&
+      hasUnsupporteKubeResourceKinds
+    )
+      return true;
+    if (fetchResourceRequestRolesAttempt.status === 'processing') return true;
+    if (isLongTerm) {
+      return !dryRunResponse?.longTermResourceGrouping?.canProceed;
+    }
```

Comment:
can you use the `longTermDisabled` here by putting that logic before this one? 🤔